packages/sdk/src/controllers/SubscriberController.ts

```diff
@@ -66,22 +66,47 @@ export class SubscriberController {
         callBack && callBack(unit as MeasurementUnit);
     }
 
+
     /**
-     * Listener on the state of the currently selected frame, if this changes, this listener will get triggered with the updates
-     * @param frameLayout Stringified object of FrameLayoutType
+     * @deprecated use `onSelectedFramesLayoutChanged` instead
+     * Listener on the state of the currently selected frames
+     * Whenever multiple frames are selected, it always will be the first selected `FrameLayoutType`
+     * @param frameLayout Stringified array of `FrameLayoutType` objects
      */
     onSelectedFrameLayoutChanged = (frameLayout: string) => {
-        const callBack = this.config.onSelectedFrameLayoutChanged;
-        callBack && callBack(JSON.parse(frameLayout));
+        const callBack = this.config.onSelectedFramesLayoutChanged;
+        const frames = JSON.parse(frameLayout);
+        callBack && callBack(frames[0]);
```

Comment:
In `sdk.ts` this is added to the subscriber methods and made available to the engine as a callback. However, this doesn't exist on the engine anymore, to me it seems that this won't work. 

This code should be moved to `onSelectedFramesLayoutChanged`, that one should invoke both callbacks instead of having the callback here. 

Maybe I'm wrong here, haven't tried linking the updated SDK to the WRS

Reply:
you are right, it didn't work. a better approach has been pushed